src/BizHawk.Client.Common/movie/tasproj/ZwinderStateManager.cs

```diff
@@ -41,24 +49,41 @@ public ZwinderStateManager(Func<int, bool> reserveCallback)
 		{
 		}
 
+		~ZwinderStateManager()
+		{
+			Dispose();
+		}
+
 		public void Engage(byte[] frameZeroState)
 		{
 			if (!_reserved.ContainsKey(0))
 			{
-				_reserved.Add(0, frameZeroState);
+				AddReserved(0, frameZeroState);
 				AddStateCache(0);
 			}
 		}
 
 		private ZwinderStateManager(ZwinderBuffer current, ZwinderBuffer recent, ZwinderBuffer gapFiller, int ancientInterval, Func<int, bool> reserveCallback)
 		{
+			SetId();
 			_current = current;
 			_recent = recent;
 			_gapFiller = gapFiller;
 			_ancientInterval = ancientInterval;
 			_reserveCallback = reserveCallback;
 		}
 		
+		private void SetId()

```

Comment:
This needs to be threadsafe -- the usual expectation in C# is that while single objects aren't threadsafe, multiple threads can independently create independent objects without conflict.  But here the internal implementation detail hits a static resource without synchronization.  Try `Interlocked.Increment`.